### PR TITLE
i#4711: Revert x86-related changes in #4677

### PR DESCRIPTION
**Pull request by sapostolakis**
Revert certain x86-related changes from #4677. In particular, we avoid restoring the arithmetic flags and (if used) the scratch register before the call to `hit_instr_count_threshold`, which might not return. 
This fix is necessary but it seems to be causing failures in `drcachesim.delay-global` and `drcacheoff.max-global` on x86-64 CI testing. These failures are not reproducible outside CI testing and are (so far) inexplicable based on manual analysis of the instrumented assembly code (see https://github.com/DynamoRIO/dynamorio/issues/4711#issuecomment-774776247). 
Thus, we are leaving x86_64 as technically broken to keep our tests green until the source of instability is found.

Issue: #4711

### CODE REVIEW

**Comment by sapostolakis** [comment]
The x86-64 tests seem to always succeed. In addition to the testing in this PR, I did a dozen manual successful runs of the ci-x86 workflow.

**Comment by abhinav92003** [comment]
If it helps, the test suite can be re-run at https://github.com/DynamoRIO/dynamorio/actions, and also from the details page for any existing CI run. I added this to the Wiki page now -- https://github.com/DynamoRIO/dynamorio/wiki/Test-Suite#continuous-integration-ci-via-github-actions.

**Comment by sapostolakis** [comment]
> If it helps, the test suite can be re-run at https://github.com/DynamoRIO/dynamorio/actions, and also from the details page for any existing CI run. I added this to the Wiki page now -- https://github.com/DynamoRIO/dynamorio/wiki/Test-Suite#continuous-integration-ci-via-github-actions.

Yeah I realized at some point last week that I could manually force runs in github actions.
Good call updating the wiki.

**Comment by derekbruening** [comment]
>￼ sapostolakis requested a review from derekbruening 5 days ago

Hmm, I missed this: because it's still marked as "Draft"?

File: clients/drcachesim/tracer/tracer.cpp
Comment (derekbruening) [inline clients/drcachesim/tracer/tracer.cpp:1508]: I'd expect a XXX/FIXME/TODO #4711 comment explaining that this needs to be done for x86 too, but this fix seems to cause instability and so we're leaving x86 as technically broken to keep our tests green until someone is able to figure out what is going on.

Comment (derekbruening) [reply clients/drcachesim/tracer/tracer.cpp:1508]: I would put this type of explanation in the commit description too, and in the (left open) issue: we're leaving x86 with buggy code b/c the fix seems to cause problems we do not yet understand.

**Review by derekbruening — COMMENTED** [review]
Please un-mark as Draft.

**Comment by derekbruening** [comment]
Oh I see: you only just now marked as Draft.  And I do see the email.  So I just missed it before.

File: clients/drcachesim/tracer/tracer.cpp
Comment (sapostolakis) [reply clients/drcachesim/tracer/tracer.cpp:1508]: Got it. I converted it back to a draft until I finished testing it.

Comment (sapostolakis) [reply clients/drcachesim/tracer/tracer.cpp:1508]: Done.

**Comment by sapostolakis** [comment]
I did a dozen CI runs and there is no failure for x86_64. So this PR should be good to go.

File: clients/drcachesim/tracer/tracer.cpp
Comment (derekbruening) [inline clients/drcachesim/tracer/tracer.cpp:1514]: style: No commented-out code.  Use #if, e.g. `#if DISABLED_FOR_BUG_4711`

**Review by derekbruening — APPROVED** [review]
Also not sure we want to hold off on merging until https://github.com/DynamoRIO/dynamorio/issues/4731#issuecomment-780127385 is fixed.

File: clients/drcachesim/tracer/tracer.cpp
Comment (sapostolakis) [reply clients/drcachesim/tracer/tracer.cpp]: Done.

**Comment by sapostolakis** [comment]
I will wait for #4731 before merging.

**Pull request merged**
